cmake/checks/check_01_compiler_features.cmake

```diff
@@ -315,6 +317,17 @@ ENDIF()
 # first see if the compiler accepts the attribute
 CHECK_CXX_SOURCE_COMPILES(
   "
+          [[deprecated]] int old_fn ();
+          int old_fn () { return 0; }
+          int (*fn_ptr)() = old_fn;
+
+          int main () {}
+  "
+  DEAL_II_COMPILER_HAS_CXX14_ATTRIBUTE_DEPRECATED
+  )
+
+CHECK_CXX_SOURCE_COMPILES(
+  "
```

Comment:
Because you changed the location of the macro, I think the __atribute__ test down here should be changed too. Otherwise the test doesn't test what we want to use.

Reply:
I changed that.